### PR TITLE
fix: fixtures moved outside the internal folder

### LINKED ISSUE #407: Move fixtures out from the internal folder
**Describe the solution you'd like**
Move the `fixtures` folder to outside `internal` and place it under root as a “sibling” to `internal`, `pkg`, `cmd` and the rest. The reason why is because `internal/` is a special directory name recognized by the go tool which will prevent one package from being imported by another unless both share a common ancestor. 

**Requirements**
Golang basic level.

### PR DESCRIPTION
**Pull request by techwithbhavya**
As mentioned in the #407 issue move the Fixtures folder to outside internal and place it under root as a "sibling" to internal, pkg, cmd, and the rest. I tried to do that.
Please review my PR for the same.
@noaabarki

### CODE REVIEW

**Comment by Abhra303** [comment]
Hello @bhavyastar,

There are some files that depend on the contents of internal folder. Say for example - 

In [files_test.go ](https://github.com/datreeio/datree/blob/1a90f04f59c9b44ef5043b31dcc923e7985dc105/bl/files/files_test.go#L39)- 
```
               args: struct{ path string }{
			path: "../../internal/fixtures/kube/pass-all.yaml",
		},
```

However, this is just one example.  You can use the search icon in your IDE to find those dependencies.

Another thing is that, your PR title contains capital letter. According to [Contributing.md](https://github.com/datreeio/datree/blob/main/CONTRIBUTING.md#short-summary), don't use capital letter and use imperative mode.

So, putting all of this together, your PR title should be similar to `refactor: move fixtures folder outside the internal folder`.

Thanks :)

**Comment by techwithbhavya** [comment]
@Abhra303 Okay, thank you so much for making me aware of this.

**Review by dimabru — CHANGES_REQUESTED** [review]
Please make the changes described in @Abhra303 comment

**Comment by techwithbhavya** [comment]
Thank you  @Abhra303, @dimabru, and @noaabarki for helping me out.😊 I made all the changes as requested. Please review my PR.

**Comment by noaabarki** [comment]
Hey @bhavyastar, thanks for your pr. Our team will review it as soon as possible.
Please note that it might take a few days 🙏🏼

**Comment by techwithbhavya** [comment]
> Hey @bhavyastar, thanks for your pr. Our team will review it as soon as possible. Please note that it might take a few days 🙏🏼

Okay, @noaabarki Thank you, ma'am.

File: bl/files/files_test.go [hunk 1]
```diff
@@ -33,11 +33,11 @@ func TestToAbsolutePath(t *testing.T) {
 }
 
 func test_existed_file() *toAbsolutePathsTestCase {
-	p, _ := filepath.Abs("../../internal/fixtures/kube/pass-all.yaml")
+	p, _ := filepath.Abs("../../fixtures/kube/pass-all.yaml")
```
Comment (noaabarki) [inline bl/files/files_test.go]: I suggest to use a const variable instead of passing the path hard-coded :)

**Review by noaabarki — CHANGES_REQUESTED** [review]
Overall look good. Please make sure you change all the references to "internal/fixtures" in all tests and also in the DEVELOPER_GUIDE.md.

File: bl/evaluation/evaluator_test.go [hunk 2]
```diff
@@ -144,7 +144,7 @@ type evaluateTestCase struct {
 }
 
 func request_evaluation_all_valid() *evaluateTestCase {
-	validFilePath := "../../internal/fixtures/kube/pass-all.yaml"
+	const validFilePath = "../../fixtures/kube/pass-all.yaml"
```
Comment (noaabarki) [inline bl/evaluation/evaluator_test.go]: Following our conversation, move this to the top(for instance, line:12) and use it in each test case. This way we won't need to repeat `../../fixtures/kube` every time.

**Review by noaabarki — CHANGES_REQUESTED** [review]
Great job, please submit this feedback and then I believe we'll be ready to go :)

File: bl/evaluation/evaluator_test.go (hunk 2 above)
Comment (techwithbhavya) [reply bl/evaluation/evaluator_test.go]: Thank you @noaabarki for suggesting me better way and helping me out. Now, it's done.

**Review by noaabarki — DISMISSED** [review]

**Review by dimabru — CHANGES_REQUESTED** [review]
Thanks a lot for this pr!
Looks great, except for the lint issues. 
Please make sure the linter is passing before we merge it.

The github action that runs the linter check has added comments in your files to make the relevant changes

**Comment by techwithbhavya** [comment]
@dimabru Okay, I will work on this linter issue.

**Comment by techwithbhavya** [comment]
@dimabru sir, Please check it once I tried to resolve that linter issue.

File: cmd/publish/main_test.go [hunk 3]
```diff
@@ -12,7 +13,7 @@ import (
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/mock"
 )
-
+const FIXTURES_PATH string = "../../fixtures/policyAsCode"
```
Comment (noaabarki) [inline cmd/publish/main_test.go:16]: Hi @bhavyastar! I believe you can run `gofmt -s -w <path_to_file>.go` to format this file.

**Review by noaabarki — CHANGES_REQUESTED** [review]

File: cmd/publish/main_test.go (hunk 3 above)
Comment (techwithbhavya) [reply cmd/publish/main_test.go:16]: Hello ma'am @noaabarki, I did the same thing, ma'am. It is still showing this. But, yes I can try again by running this command.

Comment (Abhra303) [reply cmd/publish/main_test.go:16]: I think you need to add a new line before `const FIXTURES_PATH string = "../../fixtures/policyAsCode"` i.e. -

```

const FIXTURES_PATH string = "../../fixtures/policyAsCode"
```

**Comment by noaabarki** [comment]
@bhavyastar Awesome! I'm looking forward to submitting your changes. Following our conversation, may you please resolve the conflicts?

**Comment by alexfedin** [comment]
Hi @bhavyastar,
I see that this pr was opened over a month ago and we're doing our best to promote it but unfortunately we are currently working on moving our evaluation process to the CLI, therefore we’re in a code-freeze until we merge these changes.
Your PR will be reviewed and merged once we’re out from code freeze, by the end of next week 🥳

**Comment by techwithbhavya** [comment]
Okay, Thank you. @MentalBrake

**Comment by alexfedin** [comment]
Hi @bhavyastar, Thanks for waiting, we're done with our changes and can continue with your pr.
But first, please resolve the conflicts

**Review by dimabru — CHANGES_REQUESTED** [review]
I rebased from main and pushed to this branch to update it but it looks like the output of `datree test` for anonymous users is inconsistent.

when running `DATREE_TOKEN="" make run` (on this pr)
output is `Passing policy check: 0/14`

when running the same with the latest rc release (`1.0.12-rc`): `DATREE_TOKEN="" datree test ./fixtures/**/*.yaml`
output is `Passing policy check: 0/24`

need to understand this discrepency before push

* personally I think it will be much easier to just delete this pr and start over

**Pull request closed without merging**